### PR TITLE
Add clang 8.0

### PR DESCRIPTION
**Pull request by pyhalov**
Still haven't run clang tests.
Also will add second commit to switch FF to require clang-80 and drop clang-40

### CODE REVIEW

**Comment by alarcher** [comment]
@pyhalov I am building now. I will run the tests and compile some scientific librairies.

**Comment by alarcher** [comment]
@pyhalov Need to specify /usr/bin/python3 as PYTHON in the configuration since clang expects a 64-bit python.

**Comment by alarcher** [comment]
[clang-tests.log](https://github.com/OpenIndiana/oi-userland/files/3458781/clang-tests.log)

**Comment by pyhalov** [comment]
I think it's ready to merge (as one commit).
#5184 should be merged after clang-80 is delivered.

**Comment by alarcher** [comment]
I am fine with it.

**Review by alarcher — APPROVED** [review]

**Review by Mno-hime — APPROVED** [review]
Thanks for making it happen @pyhalov & @alarcher.

File: components/developer/clang-80/patches/02-gas.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- llvm-8.0.1.src/tools/clang/lib/Driver/ToolChains/Solaris.cpp.1	2019-07-31 09:34:25.331931850 +0000
++++ llvm-8.0.1.src/tools/clang/lib/Driver/ToolChains/Solaris.cpp	2019-07-31 09:38:22.787167485 +0000
+@@ -40,7 +40,7 @@
+   for (const auto &II : Inputs)
+     CmdArgs.push_back(II.getFilename());
+ 
+-  const char *Exec = Args.MakeArgString(getToolChain().GetProgramPath("as"));
++  const char *Exec = Args.MakeArgString(getToolChain().GetProgramPath("gas"));
```
Comment (despair86) [inline components/developer/clang-80/patches/02-gas.patch:8]: pretty much all IX86 targets use the Apple integrated `as` now, this entire section is completely ignored (It even generates assembly code incompatible with any external assembler)

**Pull request merged**